File: pset2/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

int main(int argc, char *argv[]) {
  // verify valid number of arguments
  if (argc != 2) {
    printf("Usage: ./caesar k\n");
    return 1;
  }

  // initialize variables
  int key;
  char *text;
  k = atoi(argv[1]);

  // prompt user for text to be ciphered
  printf("plaintext: ");
  text = GetString();
  printf("ciphertext: ");

  // cipher the text
  for (int i = 0; i < strlen(text); i++) {
    if (isalpha(text[i]) == 0)
      printf("%c", text[i]);
    else if (islower(text[i]))
      printf("%c", (text[i] - 'a' + key) % 26 + 'a');
    else
      printf("%c", (text[i] - 'A' + key) % 26 + 'A');
  }
  printf("\n");

  // render success
  return 0;
}

File: pset6/caesar.py
from __future__ import print_function
import cs50
import sys

# encrypt a text with the caesar method
def main():
    # ensure proper usage
    if len(sys.argv) != 2:
        print("Usage: python caesar.py k")
        exit(1)

    # define and convert key to int
    key = int(sys.argv[1])

    # prompt user for text
    print("plaintext: ", end="")
    plaintext = cs50.get_string()

    print("ciphertext: ", end="")
    # iterate over plaintext, encipher each character and print it
    for i in range(len(plaintext)):
        # do not encrypt nonalphabetical character
        if plaintext[i].isalpha() == False:
            print("{}".format(plaintext[i]), end="")

        # encryption for lowercase character
        if plaintext[i].islower():
            cipher_char = chr(((ord(plaintext[i]) - ord("a")) + key) % 26 + ord("a"))
            print("{}".format(cipher_char), end="")

        # encryption for uppercase character
        if plaintext[i].isupper():
            cipher_char = chr(((ord(plaintext[i]) - ord("A")) + key) % 26 + ord("A"))
            print("{}".format(cipher_char), end="")

    # print newline
    print("")

    # render success
    return 0

# call main
if __name__ == "__main__":
    main()
